elastic/kibana
# [ResponseOps][Alerting] Provide scaling metrics for Kibana Alerting

Resolves https://github.com/elastic/kibana/issues/143189

## Summary
Added a new api to provide metrics for Kibana Alerting scaling. I implemented additional metric collection in Task Manager and updated telemetry to help determine the proper thresholds for scale up / down events. The metrics are based on the document outlined [here](https://docs.google.com/document/d/1m4BE4wKX9YKyrMa_QdLk569SXeIslTjbOYrxJSZUVkM/edit#heading=h.f518d41syfe6).


### Checklist



### To verify

- Verify that when accessing `/internal/task_manager/_background_task_utilization` you see metrics coming back in the following format:
```
{
  "@timestamp": "2022-09-28T00:00:10.000Z", // time of sample
  "process_uuid": "4777137b-d4a8-4eee-8c6f-5deb17c4a8bf", // unique identifier of the process that created these metrics
  "adhoc": {
    "created": {
      "counter": 10 // counter for number of ad hoc tasks created
    },
    "ran": {
      "service_time": {
        "actual": 523251, // total actual service time for running ad hoc tasks
        "adjusted": 600000, // total service time adjusted for polling interval
        "task_counter": 3 // ad hoc tasks counter, only increases for the lifetime of the process
      }  
    }
  },
  "recurring": {
    "ran": {
      "service_time": {
        "actual": 123251234, // total service time for running recurring tasks
        "adjusted":1476012000, // total service time adjusted for polling interval
        "task_counter": 10 // recurring tasks counter, only increases for the lifetime of the process
      }  
    }
  }
}
```
- Verify that the capacity and actual/adjusted values for adhoc and recurring tasks are added to telemetry
